### PR TITLE
Api docs

### PR DESCRIPTION
**Pull request by wzhongyuan**
## What changes were proposed in this pull request?
create MD docs for SoftSign,CosineEmbeddingCriterion,MultiLabelMarginCriterion,NarrowTable,
ReLU,SpatialCrossMapLRN,SpatialSubtractiveNormalization,Exp,Identity,SpatialZeroPadding,TanhShrink,L1Penalty

## How was this patch tested?
Tested locally for each example

### CODE REVIEW

File: docs/docs/APIdocs/layers/nn/Loss Functions/CosineEmbeddingCriterion.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+## CosineEmbeddingCriterion ##
+
+**Scala:**
+```scala
+val cosineEmbeddingCriterion = CosineEmbeddingCriterion(val margin: Double = 0.0, val sizeAverage: Boolean = true)
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Loss Functions/CosineEmbeddingCriterion.md]: If a user copy/paste this code to create an instance, he will get errors. use below instead. 
```scala
val cosineEmbeddingCriterion = CosineEmbeddingCriterion(margin = 0.0, sizeAverage = true)
```

File: docs/docs/APIdocs/layers/nn/Loss Functions/MultiLabelMarginCriterion.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
+## MultiLabelMarginCriterion ##
+
+**Scala:**
+```scala
+val multiLabelMarginCriterion = MultiLabelMarginCriterion(val sizeAverage: Boolean = true)
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Loss Functions/MultiLabelMarginCriterion.md]: remove val and types in argument list

File: docs/docs/APIdocs/layers/nn/Merge Layers/NarrowTable.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
+## NarrowTable ##
+
+**Scala:**
+```scala
+val narrowTable = NarrowTable(var offset: Int, val length: Int = 1)
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Merge Layers/NarrowTable.md]: remove val and types in argument list

File: docs/docs/APIdocs/layers/nn/Non-linear Activations/ReLU.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
+## ReLU ##
+
+**Scala:**
+```scala
+val relu = ReLU(ip: Boolean = false)
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Non-linear Activations/ReLU.md]: remove val and types in argument list

File: docs/docs/APIdocs/layers/nn/Normalization Layers/SpatialCrossMapLRN.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
+## SpatialCrossMapLRN ##
+
+**Scala:**
+```scala
+val spatialCrossMapLRN = SpatialCrossMapLRN(val size: Int = 5, val alpha: Double = 1.0, val beta: Double = 0.75, val k: Double = 1.0)
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Normalization Layers/SpatialCrossMapLRN.md]: remove val and types in argument list

File: docs/docs/APIdocs/layers/nn/Normalization Layers/SpatialSubtractiveNormalization.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
+## SpatialSubtractiveNormalization ##
+
+**Scala:**
+```scala
+val spatialSubtractiveNormalization = SpatialSubtractiveNormalization(val nInputPlane: Int = 1, var kernel: Tensor[T] = null)
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Normalization Layers/SpatialSubtractiveNormalization.md]: remove val and types in argument list

File: docs/docs/APIdocs/layers/nn/Simple Layers/L1Penalty.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
+## L1Penalty ##
+
+**Scala:**
+```scala
+val l1Penalty = L1Penalty(val l1weight: Int, val sizeAverage: Boolean = false,val provideOutput: Boolean = true)
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/nn/Simple Layers/L1Penalty.md]: remove val and types in argument list

File: docs/docs/APIdocs/layers/optim/Adagrad.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
+## Adagrad ##
+
+**Scala:**
+```scala
+val adagrad = new Adagrad(var learningRate: Double = 1e-3,
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/optim/Adagrad.md]: remove val and types in argument list

File: docs/docs/APIdocs/layers/optim/Adagrad.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+                          var learningRateDecay: Double = 0.0,
+                          var weightDecay: Double = 0.0)
+adagrad.optimize(feval, x, config, state)
```
Comment (shane-huang) [inline docs/docs/APIdocs/layers/optim/Adagrad.md]: I think usually we only put a example of how to create an instance here. Only if there's some method that user really should know, we can also put it here. 

Is the `optimize` method user really needs to know? I thought in most uses a user just create an Adagrad instance and then set it into Optimzer as `optim_method`.  

I think we shoud remove this line.

**Comment by wzhongyuan** [comment]
@shane-huang  Thanks. Please review refined above per your comments

**Pull request merged**